x/programs/runtime/imports.go

```diff
@@ -100,6 +100,21 @@ func (f Function[T, U]) call(callInfo *CallInfo, caller *wasmtime.Caller, vals [
 	return writeOutputToMemory(callInfo, results, err)
 }
 
+type InfaillibleFunction[T any] func(*CallInfo, T) RawBytes
```

Comment:
Don't need this, especially because it can still fail due to input desserialization.

Reply:
Isn't it still going to happen when we call `writeOutputToMemory` ?
I wanted to highlight that this function couldn't fail, but I guess that I just return `nil` everywhere too

Reply:
Removed this `InfaillibleFunction`, just using the `Functions` and returning `nil` for each error since the `InfaillibleFunction` was not bringing much benefits.